### DISCUSSION TITLE
When and how do the `next` versions get published to npmjs?

### DISCUSSION THREAD

**Discussion by kittaakos** [eclipse-theia/theia#12460] [Q&A] [answered]
Hi there,

How do the `next` versions get to npmjs? I see the bleeding edge [`@theia/core@1.37.0-next.22`](https://www.npmjs.com/package/@theia/core/v/1.37.0-next.22) was published five days ago, although there have been other changes on the `master` since then.

I remember the new `next` version used to be released after each commit to the `master`. How does it work now? When do the publish workflows run? I could not find the related documentation on the wiki pages.

Thank you!

**Comment by vince-fugnitto** [comment, marked as answer]
@kittaakos `next` gets published on merge to master, but we've noticed lately that it sometimes fails due to some `gh-pages` step (might actually be a github bug).

https://github.com/eclipse-theia/theia/blob/76466ccd7121f4e95f1acdd516f7cc8420f8ca84/.github/workflows/ci-cd.yml#L116-L120

Failed publishes:
- https://github.com/eclipse-theia/theia/actions/runs/4801323591/jobs/8543619267
- https://github.com/eclipse-theia/theia/actions/runs/4764933248/jobs/8473489699

**Reply by kittaakos** [reply]
> `next` gets published on merge to master

Great. Thank you!
